File: src/main/java/br/campos/poi/util/POIException.java
package br.campos.poi.util;

/**
 * Exceção personalizada para melhor tratativa
 * 
 * @author caio
 */
public class POIException extends Exception {
	private static final long serialVersionUID = 4979492878603725031L;

	/**
	 * Construtor de Exceção
	 */
	public POIException() {
		super();
	}

	/**
	 * Construtor de Exceção
	 * 
	 * @param msg mensagem de erro
	 */
	public POIException(String msg) {
		super(msg);
	}

	/**
	 * Construtor de Exceção
	 * 
	 * @param throwable origem da Exceção
	 */
	public POIException(Throwable throwable) {
		super(throwable);
	}

	/**
	 * Construtor de Exceção
	 * 
	 * @param msg       mensagem de erro
	 * @param throwable origem da Exceção
	 */
	public POIException(String msg, Throwable throwable) {
		super(msg, throwable);
	}
}

File: src/main/kotlin/br/campos/poi/util/POIException.kt
package br.campos.poi.util

/**
 * Exceção personalizada para melhor tratativa
 *
 * @author caio
 */
class POIException : Exception {

	constructor() : super()

	constructor(msg: String) : super(msg)

	constructor(throwable: Throwable) : super(throwable)

	constructor(msg: String, throwable: Throwable) : super(msg, throwable)

	companion object {
		private const val serialVersionUID: Long = 4979492878603725031L
	}
}
